Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs

```diff
@@ -146,60 +154,25 @@ private async Task AddRelatedFigures(StreetcodeContent streetcode, IEnumerable<S
         await _repositoryWrapper.RelatedFigureRepository.CreateRangeAsync(relatedFiguresToCreate);
     }
 
-    private async Task AddArts(StreetcodeContent streetcode, IEnumerable<ArtCreateDTO> arts)
+    private void AddArts(StreetcodeContent streetcode, IEnumerable<StreetcodeArtCreateUpdateDTO> arts)
     {
-        var artsToCreate = new List<StreetcodeArt>();
-
-        foreach(var art in arts)
-        {
-            var newArt = _mapper.Map<StreetcodeArt>(art);
-            newArt.Art.Image = await _repositoryWrapper.ImageRepository.GetFirstOrDefaultAsync(x => x.Id == art.ImageId);
-            newArt.Art.Image.Alt = art.Title;
-            artsToCreate.Add(newArt);
-        }
-
-        streetcode.StreetcodeArts.AddRange(artsToCreate);
+        streetcode.StreetcodeArts.AddRange(_mapper.Map<IEnumerable<StreetcodeArt>>(arts));
     }
 
-    private async Task AddTimelineItems(StreetcodeContent streetcode, IEnumerable<TimelineItemDTO> timelineItems)
+    private void AddTimelineItems(StreetcodeContent streetcode, IEnumerable<TimelineItemCreateUpdateDTO> timelineItems)
     {
-        var newContexts = timelineItems.SelectMany(x => x.HistoricalContexts).Where(c => c.Id == 0).DistinctBy(x => x.Title);
-        var newContextsDb = _mapper.Map<IEnumerable<HistoricalContext>>(newContexts);
-        await _repositoryWrapper.HistoricalContextRepository.CreateRangeAsync(newContextsDb);
-        await _repositoryWrapper.SaveChangesAsync();
-        List<TimelineItem> newTimelines = new List<TimelineItem>();
-        TimelineItem current;
-        HistoricalContext currentHistoricalContext;
-        foreach (TimelineItemDTO timelineItem in timelineItems)
-        {
-           current = _mapper.Map<TimelineItem>(timelineItem);
-           current.HistoricalContexts.Clear();
-           newTimelines.Add(current);
-           foreach (HistoricalContextDTO historicalContext in timelineItem.HistoricalContexts)
-           {
-                if (historicalContext.Id == 0)
-                {
-                    currentHistoricalContext = newContextsDb.FirstOrDefault(x => x.Title.Equals(historicalContext.Title));
-                    if(currentHistoricalContext != null)
-                    {
-                        current.HistoricalContexts.Add(currentHistoricalContext);
-                    }
-                }
-                else
-                {
-                    current.HistoricalContexts.Add(_mapper.Map<HistoricalContext>(historicalContext));
-                }
-           }
-        }
-
```

Comment:
this will break the logic, because if we will have the same newly created historical context in different timelines, we will have as much repeated in the database as was added to our timeline items.

Reply:
done